assets/model_monitoring/components/src/action_analyzer/contracts/action_detectors/low_retrieval_score_index_action_detector.py

```diff
@@ -0,0 +1,148 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+"""Low retrieval score index action detector class."""
+
+import os
+import pandas
+from typing import List
+from action_analyzer.contracts.action_detectors.action_detector import ActionDetector
+from action_analyzer.contracts.actions.action import Action
+from action_analyzer.contracts.actions.low_retrieval_score_index_action import LowRetrievalScoreIndexAction
+from action_analyzer.contracts.llm_client import LLMClient
+from action_analyzer.contracts.utils.detector_utils import (
+    extract_fields_from_debugging_info,
+    get_missed_metrics,
+    calculate_e2e_metrics,
+    get_retrieval_score,
+    get_query_intention,
+    generate_index_action_samples
+)
+from shared_utilities.constants import (
+    INDEX_SCORE_LLM_COLUMN,
+    METRICS_VIOLATION_THRESHOLD,
+    GOOD_METRICS_THRESHOLD,
+    DEFAULT_TOPIC_NAME,
+    INDEX_CONTENT_COLUMN,
+    PROMPT_COLUMN,
+    LOW_RETRIEVAL_SCORE_THRESHOLD,
+    HIGH_RETRIEVAL_SCORE_THRESHOLD,
+    INVALID_LLM_SCORE
+)
+
+
+LOW_RETRIEVAL_SCORE_QUERY_RATIO_THRESHOLD = 0.1
+
+
+class LowRetrievalScoreIndexActionDetector(ActionDetector):
+    """Low retrieval score index action detector class."""
+
+    def __init__(self,
+                 index_id: str,
+                 violated_metrics: List[str],
+                 query_intention_enabled: str) -> None:
+        """Create a low retrieval score index action detector.
+
+        Args:
+            index_id(str): the index asset id.
+            violated_metrics(List[str]): violated e2e metrics
+            query_intention_enabled(str): enable llm generated summary. Accepted values: true or false.
+            max_positive_sample_size(int): (Optional) max positive sample size in the action.
+        """
+        self.index_id = index_id
+        self.violated_metrics = violated_metrics
+        super().__init__(query_intention_enabled)
+
+    def preprocess_data(self, df: pandas.DataFrame) -> pandas.DataFrame:
+        """Preprocess the data for action detector.
+
+            1. check if all violated metrics are available. If not, call evaluate sdk to get the e2e metrics.
+            2. extract extra fields from the root span for action.
+            3. convert the dataframe from trace level to span level.
+
+        Args:
+            df(pandas.DataFrame): input pandas dataframe.
+
+        Returns:
+            pandas.DataFrame: preprocessed pandas dataframe.
+        """
+        missed_metrics = get_missed_metrics(self.violated_metrics, df.columns.tolist())
```

Comment:
Are you going to generate action for the 5 metrics? how are you going to calculate?

Reply:
I feel like we don't need it using the low_retreival score method

Reply:
make sense. Removed